pkg/auth/handler/webapp/send_oob_otp.go

```diff
@@ -0,0 +1,127 @@
+}
+
+var TemplateWebSendOOBOTPHTML = template.RegisterHTML(
+	"web/send_oob_otp.html",
+	components...,
+)
+
+type SendOOBOTPViewModel struct {
+	AlternativeSteps []viewmodels.AlternativeStep
+	OOBOTPTarget     string
+	OOBOTPCodeLength int
+	OOBOTPChannel    string
+}
+
+func ConfigureSendOOBOTPRoute(route httproute.Route) httproute.Route {
+	return route.
+		WithMethods("OPTIONS", "POST", "GET").
+		WithPathPattern("/send_oob_otp")
+}
+
+type SendOOBOTPHandler struct {
+	ControllerFactory ControllerFactory
+	BaseViewModel     *viewmodels.BaseViewModeler
+	Renderer          Renderer
+}
+
+type TriggerOOBOTPEdge interface {
+	GetOOBOTPTarget(idx int) string
+	GetOOBOTPChannel(idx int) string
+}
+
+func (h *SendOOBOTPHandler) GetData(r *http.Request, rw http.ResponseWriter, graph *interaction.Graph) (map[string]interface{}, error) {
+	data := make(map[string]interface{})
+
+	baseViewModel := h.BaseViewModel.ViewModel(r, rw)
+	viewModel := EnterOOBOTPViewModel{}
+
+	var node AuthenticationBeginNode
+	if !graph.FindLastNode(&node) {
+		panic("send_oob_otp: expected graph has node implementing AuthenticationBeginNode")
+	}
+
+	edges, err := node.GetAuthenticationEdges()
+	if err != nil {
+		return nil, err
+	}
+
+	if edge, ok := edges[0].(TriggerOOBOTPEdge); ok {
+		viewModel.OOBOTPChannel = edge.GetOOBOTPChannel(DefaultAuthenticatorIndex)
+		switch authn.AuthenticatorOOBChannel(viewModel.OOBOTPChannel) {
+		case authn.AuthenticatorOOBChannelEmail:
+			viewModel.OOBOTPTarget = mail.MaskAddress(edge.GetOOBOTPTarget(DefaultAuthenticatorIndex))
+		case authn.AuthenticatorOOBChannelSMS:
+			viewModel.OOBOTPTarget = phone.Mask(edge.GetOOBOTPTarget(DefaultAuthenticatorIndex))
+		}
+	} else {
+		panic(fmt.Errorf("send_oob_otp: unexpected edge: %T", edges[0]))
+	}
+
+	alternatives := &viewmodels.AlternativeStepsViewModel{}
+	err = alternatives.AddAuthenticationAlternatives(graph, webapp.SessionStepEnterOOBOTPAuthn)
+	if err != nil {
+		return nil, err
+	}
+	viewModel.AlternativeSteps = alternatives.AlternativeSteps
+
+	viewmodels.Embed(data, baseViewModel)
+	viewmodels.Embed(data, viewModel)
+	return data, nil
+}
+
+func (h *SendOOBOTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctrl, err := h.ControllerFactory.New(r, w)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	defer ctrl.Serve()
+
+	ctrl.Get(func() error {
+		graph, err := ctrl.InteractionGet()
+		if err != nil {
+			return err
+		}
+
+		data, err := h.GetData(r, w, graph)
+		if err != nil {
+			return err
+		}
+
+		h.Renderer.RenderHTML(w, r, TemplateWebSendOOBOTPHTML, data)
+		return nil
+	})
+
+	ctrl.PostAction("send", func() error {
+		result, err := ctrl.InteractionPost(func() (input interface{}, err error) {
+			input = &InputTriggerOOB{AuthenticatorIndex: DefaultAuthenticatorIndex}
+			return
+		})
+		if err != nil {
+			return err
+		}
+
+		result.WriteResponse(w, r)
+		return nil
+	})
```

Comment:
We need to call `handleAlternativeSteps(ctrl)` in pages with alternative steps to attach necessary action handlers.